File: app_tieng_nhat/src/main/java/com/example/app_tieng_nhat/model/StudentRegistration.java
package com.example.app_tieng_nhat.model;

import com.example.app_tieng_nhat.DTO.StudentRegistrationDTO;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Date;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
@Entity
@Table(name = "studentRegistration")
public class StudentRegistration {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String nameAndSdt;
    private String email;

    private Boolean bankCheck;
    private LocalDate regisDay;
    private String bill;
    private Boolean status;

    @ManyToOne
    @JsonBackReference("student-studentRegistration")
    @JoinColumn(name = "student_id", nullable = false)
    private Users student;

    @ManyToOne
    @JsonManagedReference("classroom-studentRegistration")
    @JoinColumn(name = "classroom_id", nullable = false)
    private ClassRoom classRoom;

    public StudentRegistrationDTO toDTO() {
        StudentRegistrationDTO dto = new StudentRegistrationDTO();
        dto.setId(this.id);
        dto.setNameAndSdt(this.nameAndSdt);
        dto.setEmail(this.email);
        dto.setBankCheck(this.bankCheck);
        dto.setRegisDay(this.regisDay);
        dto.setBill(this.bill);
        dto.setStatus(this.status);
        dto.setStudentId(this.student != null ? this.student.getId() : null);
        dto.setClassRoomId(this.classRoom != null ? this.classRoom.getId() : null);
        return dto;
    }

}

File: nihonapp2_11/lib/model/StudentRegistration.dart
class StudentRegistration {
  final int id;
  final String nameAndSdt;
  final String? email;
  final bool bankCheck;
  final DateTime regisDay;
  final String bill;
  final bool? status;
  final int classRoomId;

  StudentRegistration({
    required this.id,
    required this.nameAndSdt,
    this.email,
    required this.bankCheck,
    required this.regisDay,
    required this.bill,
    this.status,
    required this.classRoomId,
  });

  // Chuyển JSON thành đối tượng StudentRegistration
  factory StudentRegistration.fromJson(Map<String, dynamic> json) {
    return StudentRegistration(
      id: json['id'],
      nameAndSdt: json['nameAndSdt'] ?? '',
      email: json['email'], // Có thể null
      bankCheck: json['bankCheck'] ?? false,
      regisDay: DateTime.parse(json['regisDay']),
      bill: json['bill'] ?? '',
      status: json['status'], // Có thể null
      classRoomId: json['classRoomId'] ?? 0,
    );
  }

  // Chuyển đối tượng StudentRegistration thành JSON
  Map<String, dynamic> toJson() {
    return {
      "nameAndSdt": nameAndSdt,
      "regisDay": regisDay.toIso8601String(),
      "bill": bill,
      "email": email,
      "classRoom_id":
          classRoomId, // Đúng với key "classRoom_id" trong JSON yêu cầu
    };
  }
}
